### PR TITLE
ISPN-8447 Allow building with JDK 9 GA

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-8447

Requires --add-modules=java.xml.bind for karaf-maven-plugin.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -9,6 +9,9 @@ ROOT="/"
 MVN="mvn"
 
 MAVEN_OPTS="$MAVEN_OPTS -Xmx1G"
+if $JAVA_HOME/bin/java -fullversion 2>&1 | grep -q 'java full version "9' ; then
```
Comment (tristantarrant) [inline build.sh]: We need to start thinking of the post-JDK 9 world too, but for now this will do

**Comment by tristantarrant** [comment]
Using JDK 9.0.1 I get the following:

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 24.000 s
[INFO] Finished at: 2017-11-06T10:45:48+01:00
[INFO] Final Memory: 120M/400M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.karaf.tooling:karaf-maven-plugin:4.1.2:verify (validate) on project infinispan-commons: Unable to load features descriptors: Error:
[ERROR] 	javax/xml/bind/JAXBException
[ERROR] 	javax/xml/bind/JAXBException
[ERROR] 	javax/xml/bind/JAXBException
[ERROR] -> [Help 1]

**Comment by danberindei** [comment]
@tristantarrant `./build.sh clean install -DskipTests` worked for me with 9.0.1. Can you run with `bash -x` to see why the `--add-modules java.xml.bind` arguments aren't added?

**Comment by tristantarrant** [comment]
build.sh relies on JAVA_HOME being set which wasn't the case here.

**Pull request merged**